autorest/adal/token.go

```diff
@@ -650,6 +650,8 @@ func GetMSIVMEndpoint() (string, error) {
 	return msiEndpoint, nil
 }
 
+// NOTE: this only indicates if the ASE environment credentials have been set
+// which does not necessarily mean that the caller is authenticating via ASE!
```

Comment:
In track 2 we do make this assumption across languages I believe.

Reply:
Good to know, we'll need to ensure this scenario isn't broken in track 2.

Reply:
Yes, I'll bring this up with the identity crew. Did someone complain about this? Like users had these environment vars set and did not need App Service authentication?

Reply:
Nevermind, I'm reading through the issue.